src/Physics/babylon.physicsHelper.ts

```diff
@@ -481,20 +475,194 @@ module BABYLON {
 
     }
 
+
+    /***** Vortex *****/
+
+    export class PhysicsVortexEvent {
+
+        private _physicsEngine: PhysicsEngine;
+        private _originTop: Vector3 = Vector3.Zero(); // the most upper part of the cylinder
+        private _centripetalForceThreshold: number = 0.7; // at which distance, relative to the radius the centripetal forces should kick in
+        private _updraftMultiplier: number = 0.02;
+        private _tickCallback: any;
+        private _cylinder: Mesh;
+        private _cylinderPosition: Vector3 = Vector3.Zero(); // to keep the cylinders position, because normally the origin is in the center and not on the bottom
+        private _dataFetched: boolean = false; // check if the has been fetched the data. If not, do cleanup
+
+        constructor(private _scene: Scene, private _origin: Vector3, private _radius: number, private _strength: number, private _height: number) {
+            this._physicsEngine = <PhysicsEngine>this._scene.getPhysicsEngine();

```

Comment:
store the physics engine in temporary variable and check if it exists. if it doesn't, make sure to let the user know that physics needs to be enabled. If it does, set the _physicsEngine variable. this way you don't need the casting, and the user is a bit safer.

Reply:
https://github.com/bobalazek/Babylon.js/blob/ebd8ce7cc1c3fdb8eaeb94064c8e34b72e40f2df/src/Physics/babylon.physicsHelper.ts#L114

It does that already here

Reply:
oh, didn't see that :). why don't you get it there instead of in the constructor? this way it is checked when the user actually needs it.

Reply:
I think the main reason I check it there is, that few lines below I also check if any impostors exist (https://github.com/bobalazek/Babylon.js/blob/ebd8ce7cc1c3fdb8eaeb94064c8e34b72e40f2df/src/Physics/babylon.physicsHelper.ts#L120). If none, the event doesn't get fired in the first place. And of course for consistency reasons. Although, now thinking about it, I think neither of these would even be needed, because I already check for the physicsEngine in the physicsHelper constructor (https://github.com/bobalazek/Babylon.js/blob/ebd8ce7cc1c3fdb8eaeb94064c8e34b72e40f2df/src/Physics/babylon.physicsHelper.ts#L8)